From 76e5cb79535b24d78960b18046c52f4bc135fa12 Mon Sep 17 00:00:00 2001
From: xxyy <xxyy98@gmail.com>
Date: Sat, 4 Jan 2014 22:35:31 +0100
Subject: [PATCH] Add +ping -

Signed-off-by: xxyy <xxyy98@gmail.com>
---
 plugins/ping.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/plugins/ping.py b/plugins/ping.py
index 66286df..21f1882 100644
--- a/plugins/ping.py
+++ b/plugins/ping.py
@@ -19,6 +19,9 @@ def ping(inp, reply=None):
 
     args = inp.split(' ')
     host = args[0]
+    
+    if host == "-":
+        return "I am up and running."
 
     # check for a second argument and set the ping count
     if len(args) > 1:
-- 
1.8.3.msysgit.0

