From 232f7d1b7b64529fe157d2cff39ee1816de8d735 Mon Sep 17 00:00:00 2001
From: xxyy <xxyy98@gmail.com>
Date: Sat, 4 Jan 2014 22:34:42 +0100
Subject: [PATCH] Add automagic reconnect on connection loss

Signed-off-by: xxyy <xxyy98@gmail.com>
---
 core/irc.py | 17 ++++++++++++++---
 1 file changed, 14 insertions(+), 3 deletions(-)

diff --git a/core/irc.py b/core/irc.py
index 7d5896c..921c47f 100644
--- a/core/irc.py
+++ b/core/irc.py
@@ -44,9 +44,20 @@ class crlf_tcp(object):
         return socket.socket(socket.AF_INET, socket.TCP_NODELAY)
 
     def run(self):
-        self.socket.connect((self.host, self.port))
-        thread.start_new_thread(self.recv_loop, ())
-        thread.start_new_thread(self.send_loop, ())
+        while True:
+            print 'Trying to connect to IRC..'
+            try:
+                if self.socket.connect_ex((self.host, self.port)) == 0:
+                    thread.start_new_thread(self.recv_loop, ())
+                    thread.start_new_thread(self.send_loop, ())
+                    print '(Re-)connected successfully.'
+                    break;
+                else:
+                    print 'Connection failed. Retrying in 10 seconds.'
+                    time.sleep(10)
+            except socket.error as err:
+                print 'Socket error occurred. Retrying in 10 seconds.', err
+                time.sleep(10)
 
     def recv_from_socket(self, nbytes):
         return self.socket.recv(nbytes)
-- 
1.8.3.msysgit.0

