From aab431138035c9007feda8a1f6e6c2e0747f92c6 Mon Sep 17 00:00:00 2001
From: xxyy <devnull@nowak-at.net>
Date: Sat, 1 Mar 2014 23:58:58 +0100
Subject: [PATCH] Ignore some common bots' messages

Signed-off-by: xxyy <devnull@nowak-at.net>
---
 core/main.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/main.py b/core/main.py
index 0054b0a..cb690e6 100644
--- a/core/main.py
+++ b/core/main.py
@@ -156,6 +156,9 @@ def main(conn, out):
     for func, args in bot.events[inp.command] + bot.events['*']:
         dispatch(Input(conn, *out), "event", func, args)
 
+    if inp.nick in {"Eliralin", "FoxBot", "S"}:
+        return
+
     if inp.command == 'PRIVMSG':
         # COMMANDS
         if inp.chan == inp.nick:  # private message, no command prefix
-- 
1.8.3.msysgit.0

