From 8e3c3df3569a4322785adfbe4b1a250fdde077f8 Mon Sep 17 00:00:00 2001
From: xxyy <devnull@nowak-at.net>
Date: Mon, 10 Mar 2014 23:51:43 +0100
Subject: [PATCH] Allow newlines in factoids

---
 plugins/factoids.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/plugins/factoids.py b/plugins/factoids.py
index 403e6f5..1461b45 100644
--- a/plugins/factoids.py
+++ b/plugins/factoids.py
@@ -46,6 +46,8 @@ def remember(inp, nick='', db=None, notice=None):
 
     append = False
 
+    inp = string.replace(inp, "\\n", "\n")
+
     try:
         word, data = inp.split(None, 1)
     except ValueError:
-- 
1.8.3.msysgit.0

