From c6cb69bb67abad6406e7b06a4015f78b8731dce4 Mon Sep 17 00:00:00 2001
From: xxyy <xxyy98@gmail.com>
Date: Wed, 8 Jan 2014 23:38:06 +0100
Subject: [PATCH] dat :

---
 plugins/cookies.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/cookies.py b/plugins/cookies.py
index d69d258..5ae0863 100644
--- a/plugins/cookies.py
+++ b/plugins/cookies.py
@@ -3,7 +3,7 @@ from util import hook
 @hook.command
 @hook.command("share")
 @hook.command("cookies")
-def cookie(inp, me=None, nick=None)
+def cookie(inp, me=None, nick=None):
   """cookie -- Shares a cookie with you
   cookie <User> -- Shares a cookie with <person>"""
   
-- 
1.8.3.msysgit.0

