From 46985c0c436290bbd938907f10819d99583c5315 Mon Sep 17 00:00:00 2001
From: xxyy <devnull@nowak-at.net>
Date: Tue, 2 Dec 2014 15:08:50 +0100
Subject: [PATCH 12/12] Allow + prefix for factoids

---
 plugins/factoids.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/factoids.py b/plugins/factoids.py
index b5189d2..e24a1d7 100644
--- a/plugins/factoids.py
+++ b/plugins/factoids.py
@@ -114,7 +114,7 @@ def info(inp, notice=None, db=None, conn=None):
         notice("Unknown Factoid.")
 
 
-@hook.regex(r'^\? ?(.+)')
+@hook.regex(r'^[\?\+] ?(.+)')
 def factoid(inp, message=None, db=None, bot=None, action=None, conn=None, input=None):
     """?<word> -- Shows what data is associated with <word>."""
     try:
-- 
2.1.3

